src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -3232,15 +3250,18 @@ void MacroAssembler::vpcmpCCW(XMMRegister dst, XMMRegister nds, XMMRegister src,
     break;
   case neq:
     vpcmpCC(dst, nds, src, eq_cond_enc, width, vector_len);
-    vpxor(dst, dst, ExternalAddress(StubRoutines::x86::vector_all_bits_set()), vector_len, scratch_reg);
+    vallones(xtmp, vector_len);
+    vpxor(dst, xtmp, dst, vector_len);
```

Comment:
This would add extra overhead of doing vallones every time versus what we had before.

Reply:
Thanks a lot for the review.
uiCA shows that both result in 3 uops being executed if the all ones is reachable. If the external address is not reachable however, an extra `mov` instruction would need to be emitted in the current approach, leading to 1 extra uop.
Trying `_mm256_xor_si256(src, _mm256_set1_epi32(-1))` it seems that both gcc and clang use `vcmpeqd`.

Reply:
@merykitty Thanks for the explanation.